File: Objc/Recursive/Factorial/Factorial.m
//
//  Factorial.m
//  Algorithm
//
//  Created by Kuan-Wei Lin on 3/8/17.
//  Copyright © 2017 TaiwanMobile. All rights reserved.
//

#import "Factorial.h"

@implementation Factorial

- (int)factorialByNumber:(int)n
{
    if (n <= 0)
    {
        return 0;
    }
    else
    {
        return n + [self factorialByNumber:(n - 1)];
    }
}

@end

File: Swift/Recursive/Factorial.swift
//
//  Factorial.swift
//  Algorithm_Swift
//
//  Created by Kuan-Wei Lin on 3/16/17.
//  Copyright © 2017 cracktheterm. All rights reserved.
//

import UIKit

class Factorial: NSObject {

    func factorialByNumber(n:Int) -> Int
    {
        if n <= 0
        {
            return 0;
        }
        else
        {
            return n + factorialByNumber(n: n - 1);
        }
    }
}
